### ISSUE TITLE
Unable to Strong sign an individual project

### ISSUE THREAD

**Issue by kumbham** [dotnet/runtime#39328]
### Description

Steps to reproduce:
`./build.sh --projects ~/corefx/src/Microsoft.CSharp/pkg/Microsoft.CSharp.pkgproj --sign`

output:
```
  Signing Round 0: 298 files to sign.
run-detectors: unable to find an interpreter for /home/ubuntu/.nuget/packages/sn/1.0.0/sn.exe
/home/ubuntu/.nuget/packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.20213.4/tools/Sign.proj(81,5): error : Failed to strong-name sign file /home/ubuntu/corefx/artifacts/tmp/Debug/ContainerSigning/21/ref/netcoreapp3.1/netstandard.dll
/home/ubuntu/.nuget/packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.20213.4/tools/Sign.proj(81,5): error : Error during execution of signing process.
  SignToolTask execution finished.

Build FAILED.

/home/ubuntu/.nuget/packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.20213.4/tools/Sign.proj(81,5): error : Failed to strong-name sign file /home/ubuntu/corefx/artifacts/tmp/Debug/ContainerSigning/21/ref/netcoreapp3.1/netstandard.dll
/home/ubuntu/.nuget/packages/microsoft.dotnet.arcade.sdk/1.0.0-beta.20213.4/tools/Sign.proj(81,5): error : Error during execution of signing process.
    0 Warning(s)
    2 Error(s)

Time Elapsed 00:00:01.09
Build failed (exit code '1').
```

### Environment
`Ubuntu 18.04`

What's the recommended way to strong name sign the individual DLL's?

**Comment by varunmankal** [comment]
Hey @ViktorHofer, @stephentoub and @jkotas, can you help with this issue.
Thanks.

**Comment by jkotas** [comment]
Strong name signing is deprecated in .NET Core. The runtime does not check strong name signatures. I believe that the build scripts support it only on Windows and only for assemblies that need to run on .NET Framework.

What are you trying to do?

**Comment by viralmodi** [comment]
@jkotas Our product is a strong named signed library (nuget package with strong name signed dll) which has a dependency on Microsoft.CSharp. Our library will be incorporated by another application that belongs to another team in our company. Their application binaries are strong named signed. Our library has a target framework of .NET Standard 2.0. The partner team's .NET application is targeting 4.7.2 .NET framework. We have an internal requirement to build dependent binaries from source code. When we build Microsoft.CSharp from github sources, they are only Delay-Signed. We have a requirement to strong name sign them as explained in this msg. How can we achieve that?

**Comment by jkotas** [comment]
> The partner team's .NET application is targeting 4.7.2 .NET framework. 

.NET Framework comes with its own copy of Microsoft.CSharp.dll. The binary you would build yourself would not work well and it is not supported on .NET Framework.

.Microsoft.CSharp.dll built from dotnet/corefx and dotnet/runtime is for .NET Core runtime. .NET Core runtime binaries are not strong name signed (it is not needed as I have explained above). For example:
```
C:\Program Files\dotnet\shared\Microsoft.NETCore.App\3.1.3>sn -v Microsoft.CSharp.dll

Microsoft (R) .NET Framework Strong Name Utility  Version 4.0.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Microsoft.CSharp.dll does not represent a strongly named assembly
```

> How can we achieve that?

You cannot achieve that. In order to properly strong-name sign Microsoft.CSharp.dll, you would need the .NET Framework strongname private key. This strongname private key is not part of the .NET open source project.

**Comment by viralmodi** [comment]
@jkotas Thanks for responding. We have a similar issue/requirement with System.Componentmodel.Annotations. Does the above comment from you hold true for that package as well?
Just to make my understanding clear, based on your above comment, can I assume any project meant for .NET Core does not need to be strong name signed i.e. the following things are no longer true from MSFT docs:
1) You want to enable your assemblies to be referenced by strong-named assemblies, or you want to give friend access to your assemblies from other strong-named assemblies.
2) You want to centralize servicing for your app by applying publisher policy, which means the assembly must be installed in the global assembly cache.
Basically, will another strong named signed app based on .NET Core be able to incorporate (packaged with it) these non-strong name signed binaries without causing any errors?
Also, will we be able to load these non strong name signed binaries in GAC?

**Comment by jkotas** [comment]
> System.ComponentModel.Annotations. Does the above comment from you hold true for that package as well?

Yes, it holds for all assemblies that ship with .NET Framework.

```
C:\Windows\Microsoft.NET\Framework64\v4.0.30319>dir System.ComponentModel.Composition.dll
 Volume in drive C is OSDisk
 Volume Serial Number is 0A57-08C4

 Directory of C:\Windows\Microsoft.NET\Framework64\v4.0.30319

12/07/2019  02:10 AM           305,712 System.ComponentModel.Composition.dll
               1 File(s)        305,712 bytes
               0 Dir(s)  719,181,070,336 bytes free
```

>  the following things are no longer true from MSFT docs

Correct. I have submitted https://github.com/dotnet/docs/pull/19567 to clarify this.
